checks/prometheus_check.py

```diff
@@ -120,25 +124,54 @@ def parse_metric_family(self, buf, content_type):
         elif 'text/plain' in content_type:
             messages = {}  # map with the name of the element (before the labels) and the list of occurrences with labels and values
-            obj_map = {}   # map of the types of each metrics
+            obj_map = {}  # map of the types of each metrics
             obj_help = {}  # help for the metrics
-            for line in buf.splitlines():
-                self._extract_metrics_from_string(line, messages, obj_map, obj_help)
-
-            # Add type overrides:
-            for m_name, m_type in self.type_overrides.iteritems():
-                if m_type in self.METRIC_TYPES:
-                    obj_map[m_name] = m_type
-                else:
-                    self.log.debug("type override %s for %s is not a valid type name" % (m_type,m_name))
+            for metric in text_string_to_metric_families(buf):
+                metric_name = "%s_bucket" % metric.name if metric.type == "histogram" else metric.name
+                metric_type = self.type_overrides.get(metric_name, metric.type)
+                if metric_type == "untyped" or metric_type not in self.METRIC_TYPES:
+                    continue
+
+                messages[metric_name] = []
+                for sample in metric.samples:
+                    if (sample[0].endswith("_sum") or sample[0].endswith("_count")) and \
+                            metric_type in ["histogram", "summary"]:
+                        try:
+                            messages[sample[0]].append({"labels": sample[1], 'value': sample[2]})
+                        except KeyError:
+                            messages[sample[0]] = [{"labels": sample[1], 'value': sample[2]}]
+                    else:
+                        messages[metric_name].append({"labels": sample[1], 'value': sample[2]})
 
+                obj_map[metric.name] = metric_type
+                obj_help[metric.name] = metric.documentation
 
             for _m in obj_map:
-                if _m in messages or (obj_map[_m] == 'histogram' and '{}_bucket'.format(_m) in messages):
+                if _m in messages or (obj_map[_m] == 'histogram' and ('{}_bucket'.format(_m) in messages)):
                     yield self._extract_metric_from_map(_m, messages, obj_map, obj_help)
         else:
             raise self.UnknownFormatError('Unsupported content-type provided: {}'.format(content_type))
 
+    @staticmethod
+    def get_metric_value_by_labels(messages, _metric, _m, metric_suffix):
+        """
+        :param messages: dictionary as metric_name: {labels: {}, value: 10}
+        :param _metric: dictionary as {labels: {le: '0.001', 'custom': 'value'}}
+        :param _m: str as metric name
+        :param metric_suffix: str must be in (count or sum)
+        :return: value of the metric_name matched by the labels
+        """
+        metric_name = '{}_{}'.format(_m, metric_suffix)
+        unwanted_labels = ["le", "quantile"]  # are specifics keys for prometheus itself
```

Comment:
I can move `unwanted_labels` outside this scope, it's here only for clarity

Reply:
indeed, that's an unwanted allocation in a loop, let's move it as package constant